frontend/src/pages/ViewDietPlanPresetPage.tsx

```diff
@@ -0,0 +1,120 @@
+import { IDietPlan } from "@/interfaces/IDietPlan";
+import { useEffect, useState } from "react";
+import { toast } from "sonner";
+import { useParams } from "react-router";
+import { defaultDietPlan, } from "@/constants/DietPlanConsts";
+import DietPlanForm from "@/components/DietPlan/DietPlanForm";
+import { useDietPlanPresetApi } from "@/hooks/useDietPlanPresetsApi";
+import Loader from "@/components/ui/Loader";
+import ErrorPage from "./ErrorPage";
+import { Input } from "@/components/ui/input";
+import { Label } from "@/components/ui/label";
+import {  removeIdsAndVersions } from "@/utils/dietPlanUtils";
+
+export const ViewDietPlanPresetPage = () => {
+  const { id } = useParams();
+
+  const { getDietPlanPreset, updateDietPlanPreset, addDietPlanPreset } = useDietPlanPresetApi();
+
+   const [isNewPlan,setIsNewPlan] = useState<boolean>(false)
+
+    const [dietPlan, setDietPlan] = useState<IDietPlan>();
+
+    const [presetName,setPresetName]=useState<string>()
+
+    const [isLoading,setIsLoading]=useState<boolean>(false)
+
+    const [error,setError]=useState<string>()
```

Comment:
Read above comments. Also apply all these changes anywhere else in the project you may have done the same thing